qutebrowser/app.py

```diff
@@ -217,6 +217,11 @@ def _process_args(args):
     session_manager = objreg.get('session-manager')
     if not session_manager.did_load:
         log.init.debug("Initializing main window...")
+        private = config.val.content.private_browsing
+        if '--single-process' in q_app.arguments() and private is True:
+            log.init.error("Private windows are unavailable with "
+                           "the single-process process model.")
+            sys.exit(usertypes.Exit.err_init)
```

Comment:
Should probably show an error window, e.g. via `utils.error.handle_fatal_exc`. You can probably just construct an `Exception` on-the-fly.